api/v1alpha1/gateway_types.go

```diff
@@ -195,40 +192,80 @@ type GatewayStatus struct {
 
 // ListenerStatus is the status associated with each listener block.
 type ListenerStatus struct {
-	// Errors is a list of reasons why a given Listener Spec is
-	// not valid. Errors will be empty if the Spec is valid.
-	Errors []ListenerError `json:"errors"`
-
-	//
-	Address string `json:"address"`
+	// Address bound on this listener.
+	Address *ListenerAddress `json:"address"`
+	// Conditions describe the current condition of this listener.
+	Conditions []ListenerCondition `json:"conditions"`
 }
 
-// ListenerErrorReason is the type for errors associated with the
-// listener.
-type ListenerErrorReason string
+// ListenerConditionType is a type of condition associated with the listener.
+type ListenerConditionType string
 
 const (
-	// ErrListenerInvalidSpec is a generic error that is a
-	// catch all for unsupported configurations that do not match a
-	// more specific error. Implementors should try to use more
-	// specific errors instead of this one to give users and
-	// automation a more information.
-	ErrListenerInvalidSpec ListenerErrorReason = "InvalidSpec"
-	// ErrListenerBadAddress indicates the Address
-	ErrListenerBadAddress ListenerErrorReason = "InvalidAddress"
+	// ConditionInvalidListener is a generic condition that is a catch all for
+	// unsupported configurations that do not match a more specific condition.
+	// Implementors should try to use more specific condition instead of this
+	// one to give users and automation a more information.
+	ConditionInvalidListener ListenerConditionType = "InvalidListener"
+	// ConditionInvalidAddress indicates the Address is invalid.
+	ConditionInvalidAddress ListenerConditionType = "InvalidAddress"
+	// ConditionNoSuchListener indicates the listener does not exist.
+	ConditionNoSuchListener ListenerConditionType = "NoSuchListener"
```

Comment:
@robscott a `GatewayClass` condition type is not specified. What would the expected `Gateway` status contain if a user specifies a `GatewayClass` that does not exist? What would the expected `Gateway` status contain if a user specifies one or more `Routes` that do not exist?

Reply:
Great catch, my latest update adds a "NoSuchGatewayClass" condition to help with that.